Description: Use embedded versions of xstatic assets
 So we don't get to many general JS libraries in Ubuntu main.
Author: James Page <james.page@ubuntu.com>
Forwarded: no

--- a/horizon/test/settings.py
+++ b/horizon/test/settings.py
@@ -22,25 +22,25 @@ import sys
 
 import django
 from django.utils import html_parser
-import xstatic.main
-import xstatic.pkg.angular
-import xstatic.pkg.angular_cookies
-import xstatic.pkg.angular_mock
-import xstatic.pkg.bootstrap_datepicker
-import xstatic.pkg.bootstrap_scss
-import xstatic.pkg.d3
-import xstatic.pkg.font_awesome
-import xstatic.pkg.hogan
-import xstatic.pkg.jasmine
-import xstatic.pkg.jquery
-import xstatic.pkg.jquery_migrate
-import xstatic.pkg.jquery_quicksearch
-import xstatic.pkg.jquery_tablesorter
-import xstatic.pkg.jquery_ui
-import xstatic.pkg.jsencrypt
-import xstatic.pkg.qunit
-import xstatic.pkg.rickshaw
-import xstatic.pkg.spin
+import horizon.xstatic.main
+import horizon.xstatic.pkg.angular
+import horizon.xstatic.pkg.angular_cookies
+import horizon.xstatic.pkg.angular_mock
+import horizon.xstatic.pkg.bootstrap_datepicker
+import horizon.xstatic.pkg.bootstrap_scss
+import horizon.xstatic.pkg.d3
+import horizon.xstatic.pkg.font_awesome
+import horizon.xstatic.pkg.hogan
+import horizon.xstatic.pkg.jasmine
+import horizon.xstatic.pkg.jquery
+import horizon.xstatic.pkg.jquery_migrate
+import horizon.xstatic.pkg.jquery_quicksearch
+import horizon.xstatic.pkg.jquery_tablesorter
+import horizon.xstatic.pkg.jquery_ui
+import horizon.xstatic.pkg.jsencrypt
+import horizon.xstatic.pkg.qunit
+import horizon.xstatic.pkg.rickshaw
+import horizon.xstatic.pkg.spin
 
 from horizon.test import patches
 
@@ -158,49 +158,49 @@ STATICFILES_FINDERS = (
 
 STATICFILES_DIRS = [
     ('horizon/lib/angular',
-        xstatic.main.XStatic(xstatic.pkg.angular).base_dir),
+        horizon.xstatic.main.XStatic(horizon.xstatic.pkg.angular).base_dir),
     ('horizon/lib/angular',
-        xstatic.main.XStatic(xstatic.pkg.angular_cookies).base_dir),
+        horizon.xstatic.main.XStatic(horizon.xstatic.pkg.angular_cookies).base_dir),
     ('horizon/lib/angular',
-        xstatic.main.XStatic(xstatic.pkg.angular_mock).base_dir),
+        horizon.xstatic.main.XStatic(horizon.xstatic.pkg.angular_mock).base_dir),
     ('horizon/lib/bootstrap_datepicker',
-        xstatic.main.XStatic(xstatic.pkg.bootstrap_datepicker).base_dir),
+        horizon.xstatic.main.XStatic(horizon.xstatic.pkg.bootstrap_datepicker).base_dir),
     ('bootstrap',
-        xstatic.main.XStatic(xstatic.pkg.bootstrap_scss).base_dir),
+        horizon.xstatic.main.XStatic(horizon.xstatic.pkg.bootstrap_scss).base_dir),
     ('horizon/lib',
-        xstatic.main.XStatic(xstatic.pkg.d3).base_dir),
+        horizon.xstatic.main.XStatic(horizon.xstatic.pkg.d3).base_dir),
     ('horizon/lib',
-        xstatic.main.XStatic(xstatic.pkg.hogan).base_dir),
+        horizon.xstatic.main.XStatic(horizon.xstatic.pkg.hogan).base_dir),
     ('horizon/lib/font-awesome',
-        xstatic.main.XStatic(xstatic.pkg.font_awesome).base_dir),
+        horizon.xstatic.main.XStatic(horizon.xstatic.pkg.font_awesome).base_dir),
     ('horizon/lib/jasmine-1.3.1',
-        xstatic.main.XStatic(xstatic.pkg.jasmine).base_dir),
+        horizon.xstatic.main.XStatic(horizon.xstatic.pkg.jasmine).base_dir),
     ('horizon/lib/jquery',
-        xstatic.main.XStatic(xstatic.pkg.jquery).base_dir),
+        horizon.xstatic.main.XStatic(horizon.xstatic.pkg.jquery).base_dir),
     ('horizon/lib/jquery',
-        xstatic.main.XStatic(xstatic.pkg.jquery_migrate).base_dir),
+        horizon.xstatic.main.XStatic(horizon.xstatic.pkg.jquery_migrate).base_dir),
     ('horizon/lib/jquery',
-        xstatic.main.XStatic(xstatic.pkg.jquery_quicksearch).base_dir),
+        horizon.xstatic.main.XStatic(horizon.xstatic.pkg.jquery_quicksearch).base_dir),
     ('horizon/lib/jquery',
-        xstatic.main.XStatic(xstatic.pkg.jquery_tablesorter).base_dir),
+        horizon.xstatic.main.XStatic(horizon.xstatic.pkg.jquery_tablesorter).base_dir),
     ('horizon/lib/jsencrypt',
-        xstatic.main.XStatic(xstatic.pkg.jsencrypt).base_dir),
+        horizon.xstatic.main.XStatic(horizon.xstatic.pkg.jsencrypt).base_dir),
     ('horizon/lib/qunit',
-        xstatic.main.XStatic(xstatic.pkg.qunit).base_dir),
+        horizon.xstatic.main.XStatic(horizon.xstatic.pkg.qunit).base_dir),
     ('horizon/lib',
-        xstatic.main.XStatic(xstatic.pkg.rickshaw).base_dir),
+        horizon.xstatic.main.XStatic(horizon.xstatic.pkg.rickshaw).base_dir),
     ('horizon/lib',
-        xstatic.main.XStatic(xstatic.pkg.spin).base_dir),
+        horizon.xstatic.main.XStatic(horizon.xstatic.pkg.spin).base_dir),
 ]
 
-if xstatic.main.XStatic(xstatic.pkg.jquery_ui).version.startswith('1.10.'):
+if horizon.xstatic.main.XStatic(horizon.xstatic.pkg.jquery_ui).version.startswith('1.10.'):
     # The 1.10.x versions already contain the 'ui' directory.
     STATICFILES_DIRS.append(('horizon/lib/jquery-ui',
-        xstatic.main.XStatic(xstatic.pkg.jquery_ui).base_dir))
+        horizon.xstatic.main.XStatic(horizon.xstatic.pkg.jquery_ui).base_dir))
 else:
     # Newer versions dropped the directory, add it to keep the path the same.
     STATICFILES_DIRS.append(('horizon/lib/jquery-ui/ui',
-        xstatic.main.XStatic(xstatic.pkg.jquery_ui).base_dir))
+        horizon.xstatic.main.XStatic(horizon.xstatic.pkg.jquery_ui).base_dir))
 
 LOGGING = {
     'version': 1,
--- a/openstack_dashboard/settings.py
+++ b/openstack_dashboard/settings.py
@@ -22,25 +22,25 @@ import sys
 import warnings
 
 from django.utils.translation import ugettext_lazy as _
-import xstatic.main
-import xstatic.pkg.angular
-import xstatic.pkg.angular_cookies
-import xstatic.pkg.angular_mock
-import xstatic.pkg.bootstrap_datepicker
-import xstatic.pkg.bootstrap_scss
-import xstatic.pkg.d3
-import xstatic.pkg.font_awesome
-import xstatic.pkg.hogan
-import xstatic.pkg.jasmine
-import xstatic.pkg.jquery
-import xstatic.pkg.jquery_migrate
-import xstatic.pkg.jquery_quicksearch
-import xstatic.pkg.jquery_tablesorter
-import xstatic.pkg.jquery_ui
-import xstatic.pkg.jsencrypt
-import xstatic.pkg.qunit
-import xstatic.pkg.rickshaw
-import xstatic.pkg.spin
+import horizon.xstatic.main
+import horizon.xstatic.pkg.angular
+import horizon.xstatic.pkg.angular_cookies
+import horizon.xstatic.pkg.angular_mock
+import horizon.xstatic.pkg.bootstrap_datepicker
+import horizon.xstatic.pkg.bootstrap_scss
+import horizon.xstatic.pkg.d3
+import horizon.xstatic.pkg.font_awesome
+import horizon.xstatic.pkg.hogan
+import horizon.xstatic.pkg.jasmine
+import horizon.xstatic.pkg.jquery
+import horizon.xstatic.pkg.jquery_migrate
+import horizon.xstatic.pkg.jquery_quicksearch
+import horizon.xstatic.pkg.jquery_tablesorter
+import horizon.xstatic.pkg.jquery_ui
+import horizon.xstatic.pkg.jsencrypt
+import horizon.xstatic.pkg.qunit
+import horizon.xstatic.pkg.rickshaw
+import horizon.xstatic.pkg.spin
 
 from openstack_dashboard import exceptions
 
@@ -154,50 +154,50 @@ STATICFILES_FINDERS = (
 
 STATICFILES_DIRS = [
     ('horizon/lib/angular',
-        xstatic.main.XStatic(xstatic.pkg.angular).base_dir),
+        horizon.xstatic.main.XStatic(horizon.xstatic.pkg.angular).base_dir),
     ('horizon/lib/angular',
-        xstatic.main.XStatic(xstatic.pkg.angular_cookies).base_dir),
+        horizon.xstatic.main.XStatic(horizon.xstatic.pkg.angular_cookies).base_dir),
     ('horizon/lib/angular',
-        xstatic.main.XStatic(xstatic.pkg.angular_mock).base_dir),
+        horizon.xstatic.main.XStatic(horizon.xstatic.pkg.angular_mock).base_dir),
     ('horizon/lib/bootstrap_datepicker',
-        xstatic.main.XStatic(xstatic.pkg.bootstrap_datepicker).base_dir),
+        horizon.xstatic.main.XStatic(horizon.xstatic.pkg.bootstrap_datepicker).base_dir),
     ('bootstrap',
-        xstatic.main.XStatic(xstatic.pkg.bootstrap_scss).base_dir),
+        horizon.xstatic.main.XStatic(horizon.xstatic.pkg.bootstrap_scss).base_dir),
     ('horizon/lib',
-        xstatic.main.XStatic(xstatic.pkg.d3).base_dir),
+        horizon.xstatic.main.XStatic(horizon.xstatic.pkg.d3).base_dir),
     ('horizon/lib',
-        xstatic.main.XStatic(xstatic.pkg.hogan).base_dir),
+        horizon.xstatic.main.XStatic(horizon.xstatic.pkg.hogan).base_dir),
     ('horizon/lib/font-awesome',
-        xstatic.main.XStatic(xstatic.pkg.font_awesome).base_dir),
+        horizon.xstatic.main.XStatic(horizon.xstatic.pkg.font_awesome).base_dir),
     ('horizon/lib/jasmine-1.3.1',
-        xstatic.main.XStatic(xstatic.pkg.jasmine).base_dir),
+        horizon.xstatic.main.XStatic(horizon.xstatic.pkg.jasmine).base_dir),
     ('horizon/lib/jquery',
-        xstatic.main.XStatic(xstatic.pkg.jquery).base_dir),
+        horizon.xstatic.main.XStatic(horizon.xstatic.pkg.jquery).base_dir),
     ('horizon/lib/jquery',
-        xstatic.main.XStatic(xstatic.pkg.jquery_migrate).base_dir),
+        horizon.xstatic.main.XStatic(horizon.xstatic.pkg.jquery_migrate).base_dir),
     ('horizon/lib/jquery',
-        xstatic.main.XStatic(xstatic.pkg.jquery_quicksearch).base_dir),
+        horizon.xstatic.main.XStatic(horizon.xstatic.pkg.jquery_quicksearch).base_dir),
     ('horizon/lib/jquery',
-        xstatic.main.XStatic(xstatic.pkg.jquery_tablesorter).base_dir),
+        horizon.xstatic.main.XStatic(horizon.xstatic.pkg.jquery_tablesorter).base_dir),
     ('horizon/lib/jsencrypt',
-        xstatic.main.XStatic(xstatic.pkg.jsencrypt).base_dir),
+        horizon.xstatic.main.XStatic(horizon.xstatic.pkg.jsencrypt).base_dir),
     ('horizon/lib/qunit',
-        xstatic.main.XStatic(xstatic.pkg.qunit).base_dir),
+        horizon.xstatic.main.XStatic(horizon.xstatic.pkg.qunit).base_dir),
     ('horizon/lib',
-        xstatic.main.XStatic(xstatic.pkg.rickshaw).base_dir),
+        horizon.xstatic.main.XStatic(horizon.xstatic.pkg.rickshaw).base_dir),
     ('horizon/lib',
-        xstatic.main.XStatic(xstatic.pkg.spin).base_dir),
+        horizon.xstatic.main.XStatic(horizon.xstatic.pkg.spin).base_dir),
 ]
 
 
-if xstatic.main.XStatic(xstatic.pkg.jquery_ui).version.startswith('1.10.'):
+if horizon.xstatic.main.XStatic(horizon.xstatic.pkg.jquery_ui).version.startswith('1.10.'):
     # The 1.10.x versions already contain the 'ui' directory.
     STATICFILES_DIRS.append(('horizon/lib/jquery-ui',
-        xstatic.main.XStatic(xstatic.pkg.jquery_ui).base_dir))
+        horizon.xstatic.main.XStatic(horizon.xstatic.pkg.jquery_ui).base_dir))
 else:
     # Newer versions dropped the directory, add it to keep the path the same.
     STATICFILES_DIRS.append(('horizon/lib/jquery-ui/ui',
-        xstatic.main.XStatic(xstatic.pkg.jquery_ui).base_dir))
+        horizon.xstatic.main.XStatic(horizon.xstatic.pkg.jquery_ui).base_dir))
 
 COMPRESS_PRECOMPILERS = (
     ('text/scss', 'django_pyscss.compressor.DjangoScssFilter'),
